File: LeetCode-Solved-Problems/0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        files = path.split('/')
        stack = []
        for file in files:
            if file == '' or file == '.':
                continue
            if file == '..':
                if len(stack)>0:
                    stack.pop()
            else:
                stack.append(file)
        return '/'+'/'.join(stack)

File: LeetCode-Solved-Problems/0071-simplify-path/0071-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stack<string>stck;
        string ans="";
        
        for(int i=0;i<path.size();i++){
            if(path[i]=='/')continue;
                
            string temp="";
            while(i<path.size() and path[i]!='/'){
                temp+=path[i];
                i++;
            }
            if(temp==".")continue;
            else if(temp==".."){
                if(!stck.empty())
                    stck.pop();
            }
            else stck.push(temp);
            
        }
        
        while(!stck.empty()){
            ans = "/"+stck.top()+ans;
            stck.pop();
        }
        
        if(ans.size()==0)return "/";
        
        return ans;
    }
};